lib/cdo/rack/attack.rb

```diff
@@ -80,6 +80,15 @@ def update_limits
         req.write? && %r{^/v3/shared-properties/([^/]+)/[^/]+$}.match(req.path).try(:[], 1)
       end
     end
+
+    Rack::Attack.set_blacklist_map(blacklist_map)
+  end
+
+  # parse the comma-separated list of channel ids, and return them as a map
+  # which we can use to quickly check whether a given channel id is present.
+  def blacklist_map
+    channels = table_and_property_blacklist.split(',')
+    Hash[channels.map {|channel| [channel, 1]}]
```

Comment:
How about calling this blacklist_set and returning a Set?

Then you can say `Set.new(channels) `here and `@@channel_id_blacklist_set.include?(channel_id)` below.